test/base/cache.ts

```diff
@@ -217,32 +230,46 @@ export class Cache extends EventEmitter {
 			this.activeLayer = (async () => {
 				const execAsync = promisify(exec);
 
+				const targetPath = this.externalLayersPath ?? this.layersPath;
+				const gitStatusPath = this.externalLayersPath
+					? `${path.relative(await this._getGitRoot(targetPath), targetPath)}/*`
+					: 'test/simulation/cache/layers/*';
+
 				// Check git for an uncommitted layer database file
-				const { stdout: revParseStdout } = await execAsync('git rev-parse --show-toplevel');
-				const { stdout: statusStdout } = await execAsync('git status -z test/simulation/cache/layers/*', { cwd: revParseStdout.trim() });
-				if (statusStdout !== '') {
-					const layerDatabaseEntries = statusStdout.split('\0').filter(entry => entry.endsWith('.sqlite'));
-					if (layerDatabaseEntries.length > 0) {
-						const regex = /([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\.sqlite$/;
-						const match = layerDatabaseEntries[0].match(regex);
-						if (match && this.layers.has(match[1])) {
-							return this.layers.get(match[1])!;
+				try {
+					const gitRoot = await this._getGitRoot(targetPath);
+					const { stdout: statusStdout } = await execAsync(`git status -z ${gitStatusPath}`, { cwd: gitRoot });
+					if (statusStdout !== '') {
+						const layerDatabaseEntries = statusStdout.split('\0').filter(entry => entry.endsWith('.sqlite'));
+						if (layerDatabaseEntries.length > 0) {
+							const regex = /([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\.sqlite$/;
+							const match = layerDatabaseEntries[0].match(regex);
+							if (match && this.layers.has(match[1])) {
+								return this.layers.get(match[1])!;
+							}
 						}
 					}
+				} catch (error) {
+					// If git operations fail, continue to create new layer
```

Comment:
Not sure. What if `externalLayersPath` isn't inside a git repo? Shouldn't we actually throw in that case?